pkg/pillar/cmd/zfsmanager/zfsmanager.go

```diff
@@ -305,6 +309,17 @@ func deviceWatcher(ctxPtr *zfsContext) {
 		})
 	}
 
+	// Wait for directory to appear
+	for {
```

Comment:
Is this an infinite for loop inside an infinite for loop (from `deviceWatcher`)? Or is this just a loop to wait for the directory, after which this continues onwards?

Reply:
It is to wait for the directory.

What we see (but seems to be timing related) is that zfsmanager does a mkdir of /dev/zvol, but later it disappears and is recreated, but  it isn't recreated immediately. And this only happens on recent builds where we have udev. (And I've seen it happen twice on one device - second time half an hour after the device was rebooted).

Reply:
FWIW I replaced it with a fsnotify function as suggested, but it still ends up waiting forever when there is no /dev/zvol. That's ok in this goroutine since it merely needs to detect and report when a zvol appears under /dev/zvol